### PR TITLE
Add hfile loading example

### PR DESCRIPTION
**Pull request by je-ik**

### CODE REVIEW

File: euphoria-examples/build.gradle [hunk 1]
```diff
@@ -1,8 +1,35 @@
+plugins {
+  id 'java'
+  id 'com.github.johnrengelman.shadow'
```
Comment (dmvk) [inline euphoria-examples/build.gradle]: is this really needed here? java plugin is there by default and shadow plugin should be just applied

```
apply plugin: 'com.github.johnrengelman.shadow'
```

File: euphoria-examples/build.gradle [hunk 2]
```diff
@@ -1,8 +1,35 @@
...
+shadowJar {
+  relocate 'io.netty', 'cz.seznam.euphoria.example.io.netty'
+}
+
+configurations {
+  runtime.exclude group: "org.apache.hadoop"
+  runtime.exclude group: "org.apache.spark"
+  runtime.exclude group: "org.apache.flink"
+  runtime.exclude group: "org.scala-lang"
+  runtime.exclude group: "org.mortbay.jetty"
```
Comment (dmvk) [inline euphoria-examples/build.gradle]: this looks magic :)

File: euphoria-examples/src/main/java/cz/seznam/euphoria/examples/HBaseStreamer.java [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+    final Serializer<KeyValue> s = serializer.get();
+    ByteArrayOutputStream baos = new ByteArrayOutputStream();
+    for (Cell cell : res.listCells()) {
+      try {
+        s.open(baos);
+        s.serialize((KeyValue) cell);
+        s.close();
+        c.collect(baos.toByteArray());
+        baos.reset();
+      } catch (IOException ex) {
```
Comment (dmvk) [inline euphoria-examples/src/main/java/cz/seznam/euphoria/examples/HBaseStreamer.java]: ExceptionUtils?

File: euphoria-examples/src/main/java/cz/seznam/euphoria/examples/HFileLoader.java [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+        args.length >= 3, "Please specify <input_uri> <output_uri> <executor> [<tmp_dir>]");
+
+    URI input = new URI(args[0]);
+    URI output = new URI(args[1]);
+    Executor executor = Executors.createExecutor(
+        args[2],
+        KeyValue.class);
+    String tmp = "/tmp/hfileloader";
+    if (args.length > 3) {
+      tmp = args[2];
```
Comment (dmvk) [inline euphoria-examples/src/main/java/cz/seznam/euphoria/examples/HFileLoader.java]: `args[3]`?

File: euphoria-examples/src/main/java/cz/seznam/euphoria/examples/HFileLoader.java [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+        Dataset<Cell> tmp = FlatMap.of(raw)
+            .using(ResultUtil.toCells())
+            .output();
+
+        if (executor instanceof SparkExecutor) {
+          tmp = ReduceByKey.of(tmp)
+              .keyBy(Object::hashCode)
+              .reduceBy((Stream<Cell> s, Collector<Cell> ctx) -> s.forEach(ctx::collect))
+              .outputValues();
+        }
```
Comment (dmvk) [inline euphoria-examples/src/main/java/cz/seznam/euphoria/examples/HFileLoader.java]: not cool, it should at least have a good comments with explanation

File: euphoria-examples/src/main/java/cz/seznam/euphoria/examples/HFileLoader.java [hunk 6]
```diff
@@ -0,0 +1,170 @@
...
+        throw new IllegalArgumentException("Don't know how to load " + input);
+    }
+
+    ds.persist(HFileSink.newBuilder()
+        .withConfiguration(conf)
+        .withZookeeperQuorum(outputQuorum)
+        .withZnodeParent(getZnodeParent(output))
+        .withTable(table)
+        .withOutputPath(new Path(tmpDir))
+        .applyIf(ds.isBounded(),
```
Comment (dmvk) [inline euphoria-examples/src/main/java/cz/seznam/euphoria/examples/HFileLoader.java]: 👍

File: euphoria-examples/src/main/java/cz/seznam/euphoria/examples/Utils.java [hunk 7]
```diff
@@ -0,0 +1,165 @@
...
+            }
+          }
+        }
+        return builder.build();
+      }
+    }
+
+    throw new IllegalArgumentException("Invalid input URI, expected "
+        + "hbase://<zookeeper_quorum>/table/[<family>[:<qualifier>],]+");
+  }
```
Comment (dmvk) [inline euphoria-examples/src/main/java/cz/seznam/euphoria/examples/Utils.java]: is this really necessary?

File: euphoria-examples/build.gradle (hunk 1 above)
Comment (je-ik) [reply euphoria-examples/build.gradle]: :+1:

File: euphoria-examples/build.gradle (hunk 2 above)
Comment (je-ik) [reply euphoria-examples/build.gradle]: Just works. :)

File: euphoria-examples/src/main/java/cz/seznam/euphoria/examples/HBaseStreamer.java (hunk 3 above)
Comment (je-ik) [reply euphoria-examples/src/main/java/cz/seznam/euphoria/examples/HBaseStreamer.java]: Due to possible overhead I'll leave it unwrapped.

File: euphoria-examples/src/main/java/cz/seznam/euphoria/examples/HFileLoader.java (hunk 4 above)
Comment (je-ik) [reply euphoria-examples/src/main/java/cz/seznam/euphoria/examples/HFileLoader.java]: OMG! :)
:+1:

File: euphoria-examples/src/main/java/cz/seznam/euphoria/examples/HFileLoader.java (hunk 5 above)
Comment (je-ik) [reply euphoria-examples/src/main/java/cz/seznam/euphoria/examples/HFileLoader.java]: Added comment.

File: euphoria-examples/src/main/java/cz/seznam/euphoria/examples/Utils.java (hunk 7 above)
Comment (je-ik) [reply euphoria-examples/src/main/java/cz/seznam/euphoria/examples/Utils.java]: What else could I do? :)

**Pull request merged**